stuffer/s2n_stuffer.c

```diff
@@ -35,6 +35,12 @@ bool s2n_stuffer_is_valid(const struct s2n_stuffer* stuffer)
            stuffer->read_cursor <= stuffer->write_cursor;
 }
 
+bool s2n_stuffer_reservation_is_valid(const struct s2n_stuffer_reservation* reservation)
+{
+    return s2n_stuffer_is_valid(reservation->stuffer) &&
+           (reservation->write_cursor == reservation->stuffer->write_cursor);
```

Comment:
Can we also check that `length` bytes are useable?

Reply:
yep, implemented as suggested.